### PR TITLE
feat: made incidents list dynamic

### LINKED ISSUE #1540: [➕ Feature]: Creating incidents from the alert Feed
Use-case: As a user, I noticed that some alerts in the feed should be associated with a new incident. I choose those alerts, click "+Associate with an Incident," and can't assign them to the newly created incident; I can only assign them to an existing one. Even if I create a new incident in a new tab, the list of incidents in the popup won't update. If I refresh the Feed, I lose my page 😅 (covered here https://github.com/keephq/keep/issues/1541)

1) Make it possible to create a new incident from the popup
2) Make the list of incidents dynamic, so if an incident was created in the background, I should be able to choose it without updating a page.

Screenshot 2024-08-05 at 13 13 42

### PR DESCRIPTION
**Pull request by NitinPSingh**
Closes #1540 
/claim #1540 

## 📑 Description

Made the incidents list dynamic and added create incident button in AlertAssociateIncidentModal

https://github.com/user-attachments/assets/c05be923-a06a-40d8-9501-2ba0de123a0d

- [x] Make it possible to create a new incident from the popup
- [x] Make the list of incidents dynamic, so if an incident was created in the background, I should be able to choose it without updating a page.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

**Comment by Matvey-Kuk** [comment]
I see that @sergtl asked the issue #1540 to be assigned to him yesterday. Also this PR was submitted 10 minutes after the issue was claimed. I'm grateful for the contribution, but I hope you don't mind if we wait for @sergtl and won't merge it for a bit.

File: keep-ui/app/alerts/alert-associate-incident-modal.tsx [hunk 1]
```diff
@@ -61,7 +62,7 @@ const AlertAssociateIncidentModal = ({
       onClose={handleClose}
       title="Choose Incident"
       className="w-[600px]"
-    >
+    >{console.log(incidents)}
```
Comment (ozeranskii) [inline keep-ui/app/alerts/alert-associate-incident-modal.tsx:65]: Please tell me, is this information used for debugging? If so, I think it should be deleted.

**Comment by NitinPSingh** [comment]
@Matvey-Kuk, that's fine, you can wait for @sergtl. Actually, I began working on it before claiming the issue. Once the feature was completed, I then claimed the issue and submitted the PR.

**Comment by talboren** [comment]
@NitinPSingh the PR by @sergtl was merged. Thank you for the contribution!

**Pull request closed without merging**